go/worker/keymanager/worker.go

```diff
@@ -177,49 +174,43 @@ func (w *Worker) Initialized() <-chan struct{} {
 }
 
 func (w *Worker) CallEnclave(ctx context.Context, data []byte, kind enclaverpc.Kind) ([]byte, error) {
-	select {
-	case <-w.initCh:
-	default:
-		return nil, fmt.Errorf("not initialized")
-	}
-
+	// Peek into the frame/request data to extract the method.
+	var method string
 	switch kind {
 	case enclaverpc.KindNoiseSession:
-		// Handle access control as only peers on the access list can call this method.
-		peerID, ok := rpc.PeerIDFromContext(ctx)
-		if !ok {
-			return nil, fmt.Errorf("not authorized")
-		}
-
-		// Peek into the frame data to extract the method.
 		var frame enclaverpc.Frame
 		if err := cbor.Unmarshal(data, &frame); err != nil {
-			return nil, fmt.Errorf("malformed request")
-		}
-
-		// Note that the untrusted plaintext is also checked in the enclave, so if the node lied about
-		// what method it's using, we will know and the request will get rejected.
-		switch frame.UntrustedPlaintext {
-		case "":
-			// Anyone can connect.
-		default:
-			if _, privatePeered := w.privatePeers[peerID]; !privatePeered {
-				// Defer to access control to check the policy.
-				w.RLock()
-				_, allowed := w.accessList[peerID]
-				w.RUnlock()
-				if !allowed {
-					return nil, fmt.Errorf("not authorized")
-				}
-			}
+			return nil, fmt.Errorf("malformed RPC frame")
 		}
+		// Note that the untrusted plaintext is also checked in the enclave, so if the node lied
+		// about what method it's using, we will know and the request will get rejected.
+		method = frame.UntrustedPlaintext
 	case enclaverpc.KindInsecureQuery:
-		// Insecure queries are always allowed.
+		var req enclaverpc.Request
+		if err := cbor.Unmarshal(data, &req); err != nil {
+			return nil, fmt.Errorf("malformed RPC request")
+		}
+		method = req.Method
 	default:
 		// Local queries are not allowed.
 		return nil, fmt.Errorf("unsupported RPC kind")
 	}
 
+	// Handle access control.
+	switch {
+	case method == "" && kind == enclaverpc.KindNoiseSession:
+		// Anyone can connect.
```

Comment:
I assume this will be changed in a separate PR?

Reply:
Yes. A peer will be allowed to connect if it has the authorization to invoke at least one of the methods accessible through the noise session.